hotshot-state-prover/src/service.rs

```diff
@@ -403,18 +430,21 @@ pub async fn sync_state<ApiVer: StaticVersionType>(
     tracing::info!("Collected latest state and signatures. Start generating SNARK proof.");
     let proof_gen_start = Instant::now();
     let proving_key_clone = proving_key.clone();
-    let stake_table_capacity = config.stake_table_capacity;
+    let stake_table_capacity = state.config.stake_table_capacity;
+    let light_client_state = light_client_state.clone();
+    let current_stake_table_state = *current_stake_table_state;
+    let next_stake_table_state = *next_stake_table_state;
```

Comment:
given this, might as well take in owned values, we will have to copy/clone either inside or outside once anyway.

Reply:
yes ownership [574a737](https://github.com/EspressoSystems/espresso-network/pull/2960/commits/574a73722fa7058a07ffed761a84f53d965cb0f4)